cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -14,117 +14,136 @@
 * limitations under the License.
 *******************************************************************************/
 
-#include "daal/src/algorithms/covariance/oneapi/covariance_kernel_oneapi.h"
-
 #include "oneapi/dal/algo/covariance/backend/gpu/compute_kernel.hpp"
-#include "oneapi/dal/backend/interop/common_dpc.hpp"
-#include "oneapi/dal/backend/interop/error_converter.hpp"
-#include "oneapi/dal/backend/interop/table_conversion.hpp"
 
+#include "oneapi/dal/backend/primitives/lapack.hpp"
+#include "oneapi/dal/backend/primitives/reduction.hpp"
+#include "oneapi/dal/backend/primitives/stat.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
 #include "oneapi/dal/table/row_accessor.hpp"
 
 namespace oneapi::dal::covariance::backend {
 
+namespace pr = oneapi::dal::backend::primitives;
+
+using dal::backend::context_cpu;
 using dal::backend::context_gpu;
+using input_t = compute_input<task::compute>;
+using result_t = compute_result<task::compute>;
 using descriptor_t = detail::descriptor_base<task::compute>;
 
-namespace daal_covariance = daal::algorithms::covariance;
-namespace interop = dal::backend::interop;
+template <typename Float>
+auto compute_sums(sycl::queue& q,
+                  const pr::ndview<Float, 2>& data,
+                  const dal::backend::event_vector& deps = {}) {
+    const std::int64_t column_count = data.get_dimension(1);
```

Comment:
Please check if `data` has data: `ONEDAL_ASSERT(data.has_data());`
Also I consider to change naming from uninformative `data` to smth like `dataset`